### ISSUE TITLE
x86_64+arm64 fat binaries

### ISSUE THREAD

**Issue by vincentisambart** [krzysztofzablocki/Sourcery#908]
The binaries provided are currently x86_64 only.
Thanks to Rosetta, that works without any problem on M1 Macs, but it would be nice to have fat binaries supporting both arm64 and x86_64.

**Comment by krzysztofzablocki** [comment]
do you know how you get that from Swift package manager? I just recently switched us building to it

**Comment by vincentisambart** [comment]
I've never done it but it seems specify multiple architectures does the trick: https://liamnichols.eu/2020/08/01/building-swift-packages-as-a-universal-binary.html

**Comment by ferologics** [comment]
what's the status of this issue @krzysztofzablocki?

**Comment by krzysztofzablocki** [comment]
It's not as simple as just using those flags because of SwiftSyntax dylib, I didn't have time to figure it out yet, you are welcome to try figuring out how to package this

**Comment by ferologics** [comment]
I see, well I'm not so familiar with SPM packaging myself 🤷‍♂️ all I can offer is [another post](https://www.smileykeith.com/2020/12/24/swiftpm-cross-compile/) I found while researching the topic that suggests that "another option is to build once for each architecture, and then combine the binaries using lipo. Unlike the `--arch` option, this approach also works on Linux."

**Comment by vincentisambart** [comment]
Today I did some testing and indeed the binaries generated when specifying multiple `--arch` to SPM are pretty different. Doing a `swift build` per architecture then a `lipo` did seem to generated an OK binary, but once the build directory is deleted, it complains that it cannot find `Sourcery_SourceryJS.bundle`

```
SourceryJS/resource_bundle_accessor.swift:11: Fatal error: could not load resource bundle: from (...)/Sourcery/cli/bin/Sourcery_SourceryJS.bundle or (...)/Sourcery/build/x86_64-apple-macosx/release/Sourcery_SourceryJS.bundle
```

Just in case, my current diff.
```diff
diff --git a/Rakefile b/Rakefile
index eb46422..606cdc3 100644
--- a/Rakefile
+++ b/Rakefile
@@ -6,10 +6,12 @@ require 'yaml'
 require 'json'
 require 'net/http'
 require 'uri'
+require 'shellwords'
 
 BUILD_DIR = 'build/'
 CLI_DIR = 'cli/'
 VERSION_FILE = 'SourceryUtils/Sources/Version.swift'
+ARCHS = %w(x86_64 arm64)
 
 ## [ Utils ] ##################################################################
 def version_select
@@ -56,15 +58,20 @@ end
 
 task :build do
   print_info "Building project"
-  sh %Q(swift build -c release --disable-sandbox --build-path #{BUILD_DIR})
   sh %Q(rm -fr #{CLI_DIR})
   sh %Q(mkdir -p "#{CLI_DIR}bin")
   sh %Q(mkdir -p "#{CLI_DIR}lib")
   sh %Q(cp lib_InternalSwiftSyntaxParser.dylib #{CLI_DIR}lib)
   sh %Q(cp SourceryJS/Resources/ejs.js #{CLI_DIR}bin)
-  `mv #{BUILD_DIR}release/sourcery #{CLI_DIR}bin/`
-  `install_name_tool -delete_rpath @loader_path #{CLI_DIR}bin/sourcery`
-  `install_name_tool -delete_rpath /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx #{CLI_DIR}bin/sourcery`
+  developer_dir = `xcode-select -p`.chomp
+  ARCHS.each do |arch|
+    sh %Q(swift build -c release --arch #{arch} --disable-sandbox --build-path #{BUILD_DIR})
+  end
+  arch_binaries = ARCHS.map {|arch| "#{BUILD_DIR}#{arch}-apple-macosx/release/sourcery" }
+  sh %Q(lipo -create -output "#{CLI_DIR}bin/sourcery" #{arch_binaries.shelljoin})
+  `install_name_tool -delete_rpath @loader_path "#{CLI_DIR}bin/sourcery"`
+  developer_dir = `xcode-select -p`.chomp
+  `install_name_tool -delete_rpath "#{developer_dir}/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx" "#{CLI_DIR}bin/sourcery"`
   `install_name_tool -add_rpath "@executable_path/../lib" "#{CLI_DIR}bin/sourcery"`
   sh %Q(rm -fr #{BUILD_DIR})
 end
```

Also the code in Package.swift to detect if it's run in a terminal (`hookInternalSwiftSyntaxParser()`) did not seem to work properly for me when run through rake so I just commented it out.

**Comment by vincentisambart** [comment]
Trying once again `rake build` without my modifications, I still get the `Sourcery_SourceryJS.bundle` problem.

@krzysztofzablocki How are you currently building your binaries? Using `rake build` gives me a binary that searches for `Sourcery_SourceryJS.bundle`, and that bundle is deleted at the end of the `build` task with the rest of `BUILD_DIR`.

**Comment by vincentisambart** [comment]
Should be fixed by #1011
